src/query_ctx.c

```diff
@@ -171,6 +179,7 @@ bool QueryCtx_LockForCommit(void) {
 														  strlen(gc->graph_name));
 	char *error;
 	_QueryCtx_ThreadSafeContextLock(ctx);
+	ctx->internal_exec_ctx.locked_for_commit = true;
```

Comment:
Why was this line moved?
if `locked_for_commit` indicates that the graph is locked, (we're OK to modify the graph) then please revert this change.